build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -681,3 +682,191 @@ export async function setPackageDependencies(
+					`The latest dist-tag is version ${latest}, which is greater than the next dist-tag version, ${next}. Is this expected?`,
+				);
+			}
+		} else {
+			dependencyVersionMap[pkgName] = latest;
+		}
+	}
+
+	return dependencyVersionMap;
+}
+
+async function getPackagesToUpdate(
+	context: Context,
+	releaseGroup: ReleaseGroup | ReleasePackage | undefined,
+	releaseGroupFilter: ReleaseGroup | undefined,
+	log?: Logger,
+) {
+	const releaseGroupsToCheck =
+		releaseGroup === undefined // run on the whole repo
+			? [...context.repo.releaseGroups.keys()]
+			: isReleaseGroup(releaseGroup) // run on just this release group
+			? [releaseGroup]
+			: undefined;
+
+	const independentPackagesToCheck =
+		releaseGroup === undefined // run on the whole repo
+			? [...context.independentPackages] // include all independent packages
+			: isReleaseGroup(releaseGroup)
+			? [] // run on a release group so no independent packages should be included
+			: // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			  [context.fullPackageMap.get(releaseGroup)!]; // the releaseGroup argument must be a package
+
+	const packagesToUpdate: Package[] = [];
+
+	if (releaseGroupsToCheck !== undefined) {
+		for (const group of releaseGroupsToCheck) {
+			if (group === releaseGroupFilter) {
+				log?.verbose(
+					`Skipped release group ${releaseGroupFilter} because we're updating deps on that release group.`,
+				);
+				continue;
+			}
+
+			const releaseGroupRoot = context.repo.releaseGroups.get(group);
+			if (releaseGroupRoot === undefined) {
+				throw new Error(`Cannot find release group: ${group}`);
+			}
+			packagesToUpdate.push(...context.packagesInReleaseGroup(group));
+		}
+	}
+
+	if (independentPackagesToCheck !== undefined) {
+		packagesToUpdate.push(...independentPackagesToCheck);
+	}
+
+	return packagesToUpdate;
+}
+
+/**
+ * Checks the npm registry for updates for a release group's dependencies.
+ *
+ * @param context - The {@link Context}.
+ * @param releaseGroup - The release group to check. If it is `undefined`, the whole repo is checked.
+ * @param depsToUpdate - An array of packages on which dependencies should be checked.
+ * @param releaseGroupFilter - If provided, this release group won't be checked for dependencies. Set this when you are
+ * updating the dependencies on a release group across the repo. For example, if you have just released the 1.2.3 client
+ * release group, you want to bump everything in the repo to 1.2.3 except the client release group itself.
+ * @param depUpdateType - The constraint to use when deciding if updates are available.
+ * @param prerelease - If true, include prerelease versions as eligible to update.
+ * @param writeChanges - If true, changes will be written to the package.json files.
+ * @param log - A {@link Logger}.
+ * @returns An array of packages that had updated dependencies.
+ *
+ * @internal
+ */
+// eslint-disable-next-line max-params
+export async function npmCheckUpdates2(
```

Comment:
Does this need to be called npmCheckUpdates2? Is it possible to tie homegrown into it? If homegrown will change there’s no point in this then

Reply:
Good point. It could totally be npmCheckaupdatesHomegrown. I'll update.

Reply:
Homegrown as a string probably will change, but even so it's better to use the same name for the flags and the function.